ca/server.go

```diff
@@ -32,6 +34,45 @@ type APISecurityConfigUpdater interface {
 
 var _ APISecurityConfigUpdater = &Server{}
 
+// ExternalCATLSUpdater is a way to notify the server of TLS updates for the external CA
+type ExternalCATLSUpdater interface {
+	NotifyTLSChange() <-chan events.Event
+	GetCertificates() []tls.Certificate
+}
+
+// SecurityConfigTLSUpdater takes a security config and sets a watch so we can tell when TLS credentials have changed
+type SecurityConfigTLSUpdater struct {
```

Comment:
I'm wondering if this abstraction is really useful. It's only used in one place, and it seems like calling `Watch` would be clearer and lower this risk of leaking the watch (because the `cancel` return value would be ignored in that case).

Is there an upcoming change that will make this more useful?

If we keep the abstraction, can we return a `cancel` function instead of having a `Stop` method?